haproxy/assets/configuration/spec.yaml

```diff
@@ -175,8 +189,12 @@ files:
         If this parameter is >0, no datapoints will be sent until the uptime of
         haproxy reaches this many seconds.
 
-        NOTE: This only takes effect when `use_prometheus` is set to `false`.
-    - template: instances/openmetrics_legacy
+        NOTE: This only takes effect when `use_openmetrics` is set to `false`.
+    - template: instances/openmetrics
+      overrides:
+        openmetrics_endpoint.required: false
```

Comment:
Let's do something like https://github.com/DataDog/integrations-core/blob/07c10c9c98aeb2f31903e4d13367dfc3ba68fe71/cockroachdb/datadog_checks/cockroachdb/config_models/validators.py#L6-L10

Reply:
I did include some changes to the validators.py that checks for [this](https://github.com/DataDog/integrations-core/blob/steveny91/HAProxy-OpenMetricsV2/haproxy/datadog_checks/haproxy/config_models/validators.py)

It checks for the combinations of the different params needed to run either V2, V1 or legacy. I guess if you think that we should just remove the check for legacy and just do it based on V2 and V1 params then I can make that change. Hope that makes sense. :bow:

Reply:
Ah I missed that file, nice 👍